test/http-mock.ts

```diff
@@ -58,12 +58,14 @@ export function scope(basePath: BasePath, options?: nock.Options): nock.Scope {
     const { headers, method } = req;
     const url = req.options?.href;
     const result: RequestLogItem = { headers, method, url };
-    const body = req.requestBodyBuffers?.[0]?.toString();
+    const requestBody = req.requestBodyBuffers?.[0]?.toString();
 
-    if (body) {
-      try {
-        result.graphql = makeGraphqlSnapshot(body);
-      } catch (ex) {
+    if (requestBody && headers['content-type'] === 'application/json') {
+      const body = JSON.parse(requestBody);
+      const graphql = makeGraphqlSnapshot(body);
+      if (graphql) {
+        result.graphql = graphql;
+      } else {
         result.body = body;
       }
```

Comment:
I think a try catch makes sense here to be safe?  or should we simply let it crash with unhandled exception if body isn't valid json? 🤔